cocos/asset/asset-manager/downloader.ts

```diff
@@ -139,7 +139,7 @@ const downloadBundle = (nameOrUrl: string, options: Record<string, any>, onCompl
 
     const jspath = `${url}/index.${version ? `${version}.` : ''}js`;
     downloadScript(jspath, options, (err) => {
-        error = err;
+        error = err || error;
```

Comment:
I'm not sure is it necessary to keep the previous error result for each callback, Please have a look @holycanvas

Reply:
I think it's good. Because downloadJson and downloadScript can both be wrong, as long as one error occurs, the whole should be treated as an error.